editor/blocks/list/index.js

```diff
@@ -0,0 +1,40 @@
+const Editable = wp.blocks.Editable;
+const { html, prop } = wp.blocks.query;
+
+wp.blocks.registerBlock( 'core/list', {
+	title: wp.i18n.__( 'List' ),
+	icon: 'editor-ul',
+	category: 'common',
+
+	attributes: {
+		listType: prop( 'ol,ul', 'nodeName' ),
+		items: wp.blocks.query.query(
+			'li',
+			{
+				value: html()
+			}
+		)
+	},
+
+	edit( { attributes, onChange } ) {
+		const { listType = 'ol', items = [] } = attributes;
+		const value = items.map( item => {
+			return `<li>${ item.value }</li>`;
+		} ).join( '' );
+
+		return (
+			<Editable
+				tagName={ listType }
+				value={ value }
+				onChange={ onChange } />
```

Comment:
We can improve this in a future pull request, but just FYI that this isn't really doing anything. You'll probably want to set this up so that `onChange` specifically updates the `items` entries.